src/core/server/saved_objects/service/lib/repository.test.ts

```diff
@@ -79,321 +73,100 @@ import {
-    createType(MULTI_NAMESPACE_ISOLATED_TYPE, { namespaceType: 'multiple-isolated' })
-  );
-  registry.registerType(
-    createType(MULTI_NAMESPACE_CUSTOM_INDEX_TYPE, {
-      namespaceType: 'multiple',
-      indexPattern: 'custom',
-    })
-  );
-  registry.registerType(
-    createType(HIDDEN_TYPE, {
-      hidden: true,
-      namespaceType: 'agnostic',
-    })
-  );
-
-  const documentMigrator = new DocumentMigrator({
-    typeRegistry: registry,
-    kibanaVersion: KIBANA_VERSION,
-    log: loggerMock.create(),
-  });
-
-  const getMockGetResponse = (
-    {
-      type,
-      id,
-      references,
-      namespace: objectNamespace,
-      originId,
-    }: {
-      type: string;
-      id: string;
-      namespace?: string;
-      originId?: string;
-      references?: SavedObjectReference[];
-    },
-    namespace?: string | string[]
-  ) => {
-    let namespaces;
-    if (objectNamespace) {
-      namespaces = [objectNamespace];
-    } else if (namespace) {
-      namespaces = Array.isArray(namespace) ? namespace : [namespace];
-    } else {
-      namespaces = ['default'];
-    }
-    const namespaceId = namespaces[0] === 'default' ? undefined : namespaces[0];
-
-    return {
-      // NOTE: Elasticsearch returns more fields (_index, _type) but the SavedObjectsRepository method ignores these
-      found: true,
-      _id: `${
-        registry.isSingleNamespace(type) && namespaceId ? `${namespaceId}:` : ''
-      }${type}:${id}`,
-      ...mockVersionProps,
-      _source: {
-        ...(registry.isSingleNamespace(type) && { namespace: namespaceId }),
-        ...(registry.isMultiNamespace(type) && { namespaces }),
-        ...(originId && { originId }),
-        type,
-        [type]: { title: 'Testing' },
-        references,
-        specialProperty: 'specialValue',
-        ...mockTimestampFields,
-      } as SavedObjectsRawDocSource,
-    } as estypes.GetResponse<SavedObjectsRawDocSource>;
-  };
+  const registry = createRegistry();
+  const documentMigrator = createDocumentMigrator(registry);
 
-  const getMockMgetResponse = (
-    objects: Array<TypeIdTuple & { found?: boolean; initialNamespaces?: string[] }>,
-    namespace?: string
-  ) =>
-    ({
-      docs: objects.map((obj) =>
-        obj.found === false ? obj : getMockGetResponse(obj, obj.initialNamespaces ?? namespace)
-      ),
-    } as estypes.MgetResponse<SavedObjectsRawDocSource>);
-
-  expect.extend({
-    toBeDocumentWithoutError(received, type, id) {
-      if (received.type === type && received.id === id && !received.error) {
-        return { message: () => `expected type and id not to match without error`, pass: true };
-      } else {
-        return { message: () => `expected type and id to match without error`, pass: false };
-      }
-    },
-  });
   const expectSuccess = ({ type, id }: { type: string; id: string }) => {
     // @ts-expect-error TS is not aware of the extension
     return expect.toBeDocumentWithoutError(type, id);
```

Comment:
optional: this appears to be the only place where `toBeDocumentWithoutError` is called. Do you think it's worth keeping this as a custom `expect` function within `repository.test.common.ts`, or should we inline the logic here instead?